src/ert/gui/simulation/tracker_worker.py

```diff
@@ -20,16 +20,22 @@ class TrackerWorker(QObject):
     consumed_event = Signal(object)
     done = Signal()
 
-    def __init__(self, tracker: EvaluatorTracker, parent=None):
+    def __init__(
+        self,
+        event_generator_factory: Callable[
+            [], Iterator[Union[FullSnapshotEvent, SnapshotUpdateEvent, EndEvent]]
+        ],
+        parent=None,
+    ):
         super().__init__(parent)
         logger.debug("init trackerworker")
-        self._tracker = tracker
+        self._tracker = event_generator_factory
```

Comment:
Turns out you cannot do that because of `request_termination`.

Reply:
You could add `tracker_worker.py` to the list of files checked with mypy strict. Then it would have detected this defect.